src/cfn.py

```diff
@@ -125,18 +125,21 @@ def aws_stack_list():
 
 @task
 @requires_aws_stack
-def ssh(stackname, username=DEPLOY_USER):
-    public_ip = core.stack_data(stackname, ensure_single_instance=True)[0]['instance']['ip_address']
+def ssh(stackname, node="1", username=DEPLOY_USER):
+    public_ip = core.stack_data(stackname)[_node_index(node)]['instance']['ip_address']
     local("ssh %s@%s" % (username, public_ip))
 
 @task
 @requires_aws_stack
-def owner_ssh(stackname):
-    "maintainence ssh. uses the pem key and the bootstrap user to login."
-    public_ip = core.stack_data(stackname, ensure_single_instance=True)[0]['instance']['ip_address']
+def owner_ssh(stackname, node="1"):
+    "maintenance ssh. uses the pem key and the bootstrap user to login."
+    public_ip = core.stack_data(stackname)[_node_index(node)]['instance']['ip_address']
     # -i identify file
     local("ssh %s@%s -i %s" % (BOOTSTRAP_USER, public_ip, stack_pem(stackname)))
 
+def _node_index(node_argument):
+    assert node_argument.isdigit, "You must pass a node number (greater or equal than 1)"
```

Comment:
There is a function in `buildercore.utils` called `die` with a similar structure to the `assert` statement that won't be compiled away if optimisations are ever turned on. It raises a `RuntimeError` but perhaps should raise an `AssertionError`.